drivers/sensor/grow_r502a/grow_r502a.h

```diff
@@ -108,56 +112,56 @@
 #define R502A_PKG_LEN_IDX 7
 #define R502A_CC_IDX 9 /* Confirmation code index*/
 
+#define R502A_COMMON_ACK_LEN 12
+
 #define R502A_STARTCODE_LEN 2
 #define R502A_ADDRESS_LEN 4
 #define R502A_PKG_LEN	2
 #define R502A_CHECKSUM_LEN 2 /* Checksum length in uart packages*/
 #define R502A_HEADER_LEN 9
 
+/*Default data package length, can be 32 or 64 or 128 or 256*/
+#define R502A_DATA_PKT_SIZE 128
+
 #define R502A_CHAR_BUF_1 1
 #define R502A_CHAR_BUF_2 2
+#define R502A_CHAR_BUF_TOTAL 2
+
 #define R502A_CHAR_BUF_SIZE 384 /* Maximum size of characteristic value buffer*/
 #define R502A_TEMPLATE_SIZE 768 /* Maximum size of template, twice of CHAR_BUF*/
-#define R502A_MAX_BUF_SIZE 779 /*sum of checksum, header and template sizes*/
-#define R502A_BUF_SIZE 64
+
+#define R502A_TEMPLATE_MAX_SIZE (R502A_CHAR_BUF_TOTAL * R502A_TEMPLATE_SIZE)
+#define R502A_MAX_BUF_SIZE  R502A_DATA_PKT_SIZE + R502A_COMMON_ACK_LEN
+
 #define R502A_TEMPLATES_PER_PAGE 256
 #define R502A_TEMP_TABLE_BUF_SIZE 32
 #define R502A_DELETE_COUNT_OFFSET 1
 
 #define R502A_DELAY 200
 #define R502A_RETRY_DELAY 5
 
-#define LED_CTRL_BREATHING 0x01
-#define LED_CTRL_FLASHING 0x02
-#define LED_CTRL_ON_ALWAYS 0x03
-#define LED_CTRL_OFF_ALWAYS 0x04
-#define LED_CTRL_ON_GRADUALLY 0x05
-#define LED_CTRL_OFF_GRADUALLY 0x06
-
-#define LED_SPEED_HALF 0x50
-#define LED_SPEED_FULL 0xFF
-
-#define LED_COLOR_RED 0x01
-#define LED_COLOR_BLUE 0x02
-#define LED_COLOR_PURPLE 0x03
-
-struct led_params {
-	uint8_t ctrl_code;
-	uint8_t color_idx;
-	uint8_t speed; /* Speed 0x00-0xff */
-	uint8_t cycle; /* Number of cycles | 0-infinite, 1-255 */
+#define R502A_LED_INPUT_PARAMS	4
+union r502a_led_params {
+	struct {
+		uint8_t ctrl_code;
+		uint8_t color_idx;
+		uint8_t speed; /* Speed 0x00-0xff */
+		uint8_t cycle; /* Number of cycles | 0-infinite, 1-255 */
+	} __packed;
+	uint32_t buf;
```

Comment:
don't know the need for buf and union here... only place use of buf is attr SENSOR_ATTR_R502A_DEVICE_LED, memcpy will do the job